test/bigtest/tests/circulation-rules-test.js

```diff
@@ -562,7 +566,102 @@ describe('CirculationRules', () => {
           });
 
           it('the library code should be added to the editor value', () => {
-            expect(circulationRules.editor.value).to.be.equal(`c ${libraries[0].code} `);
+            expect(circulationRules.editor.value).to.be.equal(`c ${replacer(libraries[0].code)} `);
+          });
+        });
+      });
+    });
+  });
+
+  describe('choosing criteria locating criteria - location', () => {
+    const rulesEditor = circulationRules.editor;
+    const editorHints = rulesEditor.hints;
+
+    beforeEach(async function () {
+      await rulesEditor.setValue('s ');
+    });
+
+    it('should display hints with list to select location codes', () => {
+      expect(editorHints.arePresent).to.be.true;
+    });
+
+    it('should display hints with correct header', () => {
+      expect(editorHints.header.isPresent).to.be.true;
+      expect(editorHints.header.text).to.be.equal('Select location');
+    });
+
+    it('should display three hints sections', () => {
+      expect(editorHints.sectionsCount).to.be.equals(4);
+    });
+
+    it('should display hints with correct subheaders', () => {
+      const subheaderText = sectionNumber => editorHints.sections(sectionNumber).subheader.text;
```

Comment:
I do not see much value in this function as below you use `editorHints.sections(2).items()` but do not have a function for it. I think it is useful to have just function to retrieve the section and then reuse it in other places. Place it in the top of the file.

For example here.
```
const getEditorHintSection = sectionNumber => editorHints.sections(sectionNumber);
....

it('..', () => {
  ...
  getEditorHintSection(0).subheader.text.to.be.equal('Institution')
}